### DISCUSSION TITLE
Canvas Help

### DISCUSSION THREAD

**Discussion by DtueurJ** [fsprojects/Avalonia.FuncUI#285] [Q&A]
I am currently trying to do an app using the drawingApp example. Is there anyway to save the canvas as an image in any format ?

**Comment by JaggerJo** [comment]
https://github.com/fsprojects/Avalonia.FuncUI/blob/e4b5503559908d18ddc5cac28a87741dea35c3fd/src/Examples/Component%20Examples/Examples.DrawingApp/Program.fs#L47-L62

**Reply by DtueurJ** [reply]
I have a little issue with your save function. In the "path" variable, at line 59, the format of the datetime given create an error. I'm on Windows and a format "day/month/year" create an error. Unless this, it works perfectly

**Reply by JaggerJo** [reply]
Ahh, it depends on your locale. I'm in Germany so dates look like this: `21.03.2023`

**Comment by DtueurJ** [comment]
In my project, i was trying to load an image into my canvas but i can't figure how to do it ?

**Reply by JaggerJo** [reply]
A Canvas in Avalonia is not a low level drawing canvas by design. You can do some drawing in it, but it's not a low level drawing surface. If you need that I recommend taking a look at [Custom Drawing Operations](https://reference.avaloniaui.net/api/Avalonia.Media/DrawingContext/D518B33B). 

However, you can definitely add an Image to a Canvas.

Option A: Set the background of the canvas to your image, using an [ImageBrush](https://reference.avaloniaui.net/api/Avalonia.Media/ImageBrush/).

Option B: Place an Image control in `Canvas.children`. You can position it using `top/left/right/bottom` attached properties.
